### ISSUE TITLE
Bug: Union Type Results fails to Serialize/Deserialize

### ISSUE THREAD

**Issue by schang412** [dapper91/pydantic-xml#25]
# Problem
Currently, if I write: 
```python
import pydantic_xml as pyxml
from typing import Literal
from typing import Union


class Person(pyxml.BaseXmlModel):
	name: str = pyxml.attr()


class CEO(Person):
	position: Literal['CEO'] = pyxml.attr()


class CTO(Person):
	position: Literal['CTO'] = pyxml.attr()


class Company(pyxml.BaseXmlModel):
	employee_of_month: Union[CEO, CTO] = pyxml.wrapped('employee_of_month', pyxml.element(tag='person'), discriminator='position')

a = Company(employee_of_month=CEO(name='Vincent', position='CEO'))
print(a.to_xml(encoding='UTF-8').decode())
```

I get the error message: `TypeError: Object of type CEO is not XML serializable`. 

I expect that this should not fail, since `CEO` is one of the valid options. Likewise the following XML should be successfully deserialized:
```xml
<Company>
  <employee_of_month>
    <person name="Vincent" position="CEO"/>
  </employee_of_month>
</Company>
```

However, it results in validation error: `pydantic.error_wrappers.ValidationError: 4 validation errors for Company`

# Justification

This feature is supported by native pydantic:
```python
a = Company(employee_of_month=CEO(name='Vincent', position='CEO'))
a_str = a.json()
a_str = a_str.replace('CEO', 'CTO')
b = Company.parse_raw(a_str)
```

This feature is desirable in the scenario that an extra field for certain people:
```python
class CEO(Person):
	position: Literal['CEO'] = pyxml.attr()


class CTO(Person):
	position: Literal['CTO'] = pyxml.attr()


class PublicRelationsDirector(Person):
	position: Literal['PR'] = pyxml.attr()
	public_email: str = pyxml.attr(name='email')
```
Support for this will also allow for more robust deserialization. For example, consider the following XML: 
```xml
<people>
	<person name='Steven' position='engineer'>
		<product>My App</product>
	</person>
	<person name='Phil' position='engineer'>
		<project>Infrastructure Upgrade</project>
	</person>
	<person name='Ryan' position='engineer'>
		<product>My App</product>
	</person>
</people>
```
With this feature, the discriminator could be used to sort between engineers working on internal infrastructure and user facing product.

**Comment by dapper91** [comment]
@schang412 Hi,

Due to implementation details `Union` types are not yet supported.

**Comment by dapper91** [comment]
Union type support implemented in version 0.6.0.
